piper.in

```diff
@@ -7,6 +7,10 @@ import sys
 gi.require_version('Gio', '2.0')
 gi.require_version('Gtk', '3.0')
 from gi.repository import Gio, Gtk
+gtk_version = Gtk.get_major_version(), Gtk.get_minor_version()
+if gtk_version < (3, 22):
```

Comment:
it'd be possible somewhere with a script (most likely) but meson is build-time verification vs what we need is run-time verification. That's just a general problem of anything non-compiled, since you don't link to specific dependencies, a lot of the existing checks cannot be done.

in other words: it doesn't matter if the build system doesn't have gtk 3.22, it only matters where piper is run